scopesim/commands/user_commands.py

```diff
@@ -299,13 +346,13 @@ def add_packages_to_rc_search(local_path, package_list):
         A list of the package names to add
 
     """
-
+    plocal_path = patch_fake_symlinks(Path(local_path))
     for pkg in package_list:
-        pkg_dir = os.path.abspath(os.path.join(local_path, pkg))
-        if not os.path.exists(pkg_dir):
+        pkg_dir = plocal_path / pkg
+        if not pkg_dir.exists():
             # todo: keep here, but add test for this by downloading test_package
             # raise ValueError("Package could not be found: {}".format(pkg_dir))
-            logging.warning("Package could not be found: {}".format(pkg_dir))
+            logging.warning(f"Package could not be found: {pkg_dir}")
```

Comment:
Logging prefers "lazy formatting" with an additional argument, because depending on the logging level, a logging message might not need to be evaluated. If advanced formatting is used, the string expression is always evaluated, regardless if it is going anywhere or not. This can impact performance, _if_ it is something that's called a lot.

The lazy version would be ```logging.warning("Package could not be found: %s", pkg_dir)```

In this particular case however, I think it's fine to keep the advanced formatting, because a) it's a function that is probably not called too often, and b) a warning level logging message will most likely be evaluated anyway...

Reply:
Thanks, I did not know that about logging, it makes sense. I'll keep that in mind.